src/main/golo/adapters.golo

```diff
@@ -0,0 +1,160 @@
+# ............................................................................................... #
+#
+# Copyright 2012-2015 Institut National des Sciences Appliquées de Lyon (INSA-Lyon)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# ............................................................................................... #
+
+----
+This module defines an AdapterFabric helper.
+----
+module gololang.Adapters
+
+struct adapter = {
+  definition
+}
+
+----
+adapter augmentations
+----
+augment gololang.Adapters.types.adapter {
+
+  ----
+  Defines interface(s) of the Adapter:
+  Parameter(s): a tuple of strings
+  Returns the adapter
+
+  Use:
+
+      let runnerAdapter = Adapter()
+        : interfaces(["java.io.Serializable", "java.lang.Runnable"])
+  ----
+  function interfaces = |this, interfacesTuple| {
+    this: definition(): put("interfaces", interfacesTuple)
+    return this
+  }
+
+  ----
+  Provides a method implementation
+  Parameter(s): method name, anonymous function
+  Returns the adapter
+
+  Use:
+
+      let result = array[1,2,3]
+
+      let runnerAdapter = Adapter()
+        : interfaces(["java.io.Serializable", "java.lang.Runnable"])
+        : implements("run", |this| {
+            for (var i = 0, i < result: length(), i = i + 1) {
+              result: set(i, result: get(i) + 10)
+            }
+          })
+  ----
+  function implements = |this, methodName, closure| {
+    #if this: definition(): get("implements") is null {
+    #  this: definition(): put("implements", map[])
+    #}
+    this: definition(): get("implements"): put(methodName, closure)
+    return this
+  }
+
+  ----
+  Specifies the name of the parent class (`java.lang.Object` by default cf `Adapter` constructor method)
+  Parameter(s): class name
```

Comment:
Specifies the name of the parent class (`java.lang.Object` by default cf `Adapter` constructor method)
  Parameter(s): class name
  Returns the adapter

  Usage:

      let arrayListAdapter = Adapter()
        : extends("java.util.ArrayList")